web/components/roomConfig/GitlabRepoConfig.tsx

```diff
@@ -0,0 +1,167 @@
+    const [results, setResults] = useState<GitLabRepoConnectionProjectTarget[]|null>(null);
+    const [instances, setInstances] = useState<GitLabRepoConnectionInstanceTarget[]|null>(null);
+    const [debounceTimer, setDebounceTimer] = useState<number>(null);
+    const [currentProjectPath, setCurrentProjectPath] = useState<string|null>(null);
+
+    const searchFn = useCallback(() => {
+        api.getConnectionTargets<GitLabRepoConnectionTarget>(EventType, filter).then(res => {
+            if (!filter.instance) {
+                setInstances(res);
+                if (res[0]) {
+                    setFilter({instance: res[0].name});
+                }
+            } else {
+                setResults(res as GitLabRepoConnectionProjectTarget[]);
+            }
+        });
+    }, [api, filter]);
+
+    const updateSearchFn = useCallback((evt: InputEvent) => {
+        setFilter({...filter, search: (evt.target as HTMLInputElement).value });
+    }, [filter, setFilter]);
+
+    useEffect(() => {
+        if (debounceTimer) {
+            clearTimeout(debounceTimer);
+        }
+        // Browser types
+        setDebounceTimer(setTimeout(searchFn, 500) as unknown as number);
+        // Things break if we depend on the thing we are clearing.
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [setDebounceTimer, searchFn, filter, instances]);
+
+    useEffect(() => {
+        const hasResult = results?.find(n => n.name === filter.search);
+        if (hasResult) {
+            onPicked(hasResult.state);
+            setCurrentProjectPath(hasResult.state.path);
+        }
+    }, [onPicked, results, filter, setCurrentProjectPath]);
+
+
+
+    return <div>
+        {instances === null && <p> Loading GitLab instances. </p>}
+        {instances?.length === 0 && <p> You are not logged into any GitLab instances. </p>}
+        <InputField visible={instances?.length > 0} label="GitLab Instance" noPadding={true}>
+            {/*TODO: on change required here*/}
+            <select>
+                {instances?.map(i => <option key={i.name}>{i.name}</option>)}
+            </select>
+        </InputField>
+        <InputField visible={instances?.length > 0} label="Project" noPadding={true}>
+            <small>{currentProjectPath ?? ""}</small>
+            <input onChange={updateSearchFn} list="gitlab-projects" type="text" />
+            <datalist id="gitlab-projects">
+                {results?.map(i => <option path={i.state.path} value={i.name} key={i.name} />)}
+            </datalist>
+        </InputField>
+    </div>;
+}
+
+const ConnectionConfiguration: FunctionComponent<ConnectionConfigurationProps<unknown, GitLabRepoResponseItem, GitLabRepoConnectionState>> = ({api, existingConnection, onSave, onRemove }) => {
+    const [ignoredHooks, setIgnoredHooks] = useState<string[]>(existingConnection?.config.ignoreHooks || []);
+    const setIgnoredHook = useCallback(evt => {
```

Comment:
This function had me confused for a while. I expected it to be looking at `event.target.checked` but you use the component's state.

That's fine, but maybe rename the function.

```suggestion
    const toggleIgnoredHook = useCallback(evt => {
```